docs/xshareclasses.md

```diff
@@ -437,7 +437,28 @@ behavior, which can improve the performance of class loading from the shared cla
 
 :   Uses a persistent cache. The cache is created on disk, which persists beyond operating system restarts. Non-persistent and persistent caches can have the same name. On AIX, you must always use the `persistent` suboption when you run utilities such as `destroy` on a persistent cache.
 
-:   :fontawesome-solid-pencil:{: .note aria-hidden="true"} **Note:** Persisent caches are not supported on z/OS systems. z/OS supports only non-persistent caches.
+:   :fontawesome-solid-pencil:{: .note aria-hidden="true"} **Notes:**
+
+:    - On 31-bit VM on all versions of z/OS, the shared cache is memory mapped within the 0-2 GB address range. The size of the persistent shared class cache is limited by the system limit `MAXMMAPAREA`. For more information about the suggested value for the `MAXMMAPAREA` limit, see [Setting resource limits (z/OS)](https://www.eclipse.org/openj9/docs/configuring/#setting-resource-limits-zos).
+     - On 64-bit VM, the shared cache is mapped as follows:
+         - On z/OS version 2.3 and earlier, the persistent shared cache is memory mapped within the 0-2 GB address range. The size of the persistent shared class cache is limited by the system limit `MAXMMAPAREA`.
+         - On z/OS version 2.4 and later, the persistent shared cache is mapped above the 2 GB address range. The size of persistent shared class cache is limited by the `MAXSHARE` value within the `SMFLIMxx` member of `SYS1.PARMLIB`. For the above 2 GB memory map, the following machine configuration is required:
+
+              1. Add the following line in the SMFLIMxx member of SYS1.PARMLIB.
+
+             ```
+              REGION JOBNAME(*) MAXSHARE(2147483647)
```

Comment:
I worry about the recommendation that this specific line is **required**.  We are essentially saying to allow maximum shared pages for all jobs on the system, which may / may not be desirable.  This should be customized by the user, based on their needs.  Should we consider a suggested minimum size like we had for MAXMMAPAREA instead?

Reply:
I believe this value corresponds to the maximum allowed shared class cache that can be allocated using map64. May be we should put a value that is required for default shared class cache size and mention that size of maximum shared class cache is directly controlled by the value put in this parmlib?


Surely there would have been a rationale behind setting MAXMMAPAREA value to 40960 with respect to SCC. Just for knowledge, @hangshao0 , Joran do you know?

Reply:
Well, from https://www.ibm.com/docs/en/zos/2.5.0?topic=values-statements-parameters-smflimxx, 2147483647 is the maximum value allowed.  I agree, a size that is closer to what the typical SCC size (default x 2) would be more amenable as a starting point.  MAXMMAPAREA is in 4k pages, so 40960 ==> 160MB?

> MAXSHARE(nn)
    Specifies the maximum of source and target shared pages that can be used at one time by problem state callers using the IARVSERV SHARE services. This corresponds to IEFUSI parameter word 7, subword 4 output field. The maximum value for source and target shared pages is 2147483647.

Reply:
To keep it inline with what we recommended with `MAXMMAPAREA` (160 MB), may be `167772160` would be a good starting point for recommended value of MAXSHARE ?

Reply:
So, should I replace `2147483647` with `167772160` in `REGION JOBNAME(*) MAXSHARE(2147483647)`?

Reply:
@hangshao0 @r30shah - Any update on this? What change is required in this content? Thanks!

Reply:
> Any update on this? What change is required in this content? Thanks!

Rahil is reaching out to the z/OS team on this. I believe he will update once we get answers from the z/OS expert.

Reply:
I am fine with either 1GB or 2GB limit. Does the value of `MAXSHARE(1073741823)` mean bytes or pages ? If it is pages, it is much more than 1GB.

Reply:
It is actually a number of pages. Worked offline with Hang and also checked with z/OS expert. Based on that, I think we should use `MAXSHARE(524288)` that would allow us to use 2GB SCC which is the maximum size we can have with SCC. 

@hangshao0 , @joransiu  would appreciate your comment.

Reply:
I am fine with `MAXSHARE(524288)`